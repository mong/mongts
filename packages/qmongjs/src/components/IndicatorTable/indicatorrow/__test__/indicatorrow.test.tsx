import { render } from "@testing-library/react";
import { QueryParamProvider } from "use-query-params";

import { IndicatorRow } from "../";
import { NextAdapter } from "next-query-params";
import mockRouter from "next-router-mock";
import { createDynamicRouteParser } from "next-router-mock/dynamic-routes";
import {
  buildIndicator,
  buildDescription,
} from "../../../../test/test_data/dataBuilder";

jest.mock("next/router", () => require("next-router-mock"));
// This is needed for mocking 'next/link':
jest.mock("next/dist/client/router", () => require("next-router-mock"));

mockRouter.useParser(
  createDynamicRouteParser([
    // These paths should match those found in the `/pages` folder:
    "/alle/[tab].js",
    "/[register]/[tab].js",
  ])
);
beforeEach(() => {
  mockRouter.setCurrentUrl("/alle/sykehus");
});

it("renders with low achievement level and not complete year", () => {
  const data = buildIndicator({
    denominator: 2775,
    var: 0.1081,
    level_direction: 0,
    level_green: 0.05,
    level_yellow: 0.1,
  });
  const descr = buildDescription({
    id: "barnediabetes_hba1c_ge_9",
    title: "HbA1c â‰¥ 9,0 %",
    short_description: "Andel pasienter med HbA1c > 9,0",
    rname: "barnediabetes",
  });
  const { container } = render(
    <QueryParamProvider adapter={NextAdapter}>
      <table>
        <tbody>
          <IndicatorRow
            context={{ context: "caregiver", type: "ind" }}
            description={descr}
            indicatorData={[data]}
            treatmantYear={2019}
          />
        </tbody>
      </table>
    </QueryParamProvider>
  );
  expect(container).toMatchSnapshot();
});

it("renders with no achievement level on indicator", () => {
  const data = buildIndicator({
    denominator: 2676,
    var: 0.1988,
    level_direction: null,
    level_green: null,
    level_yellow: null,
  });
  const descr = buildDescription({
    id: "barnediabetes_hba1c_lt_7",
    title: "HbA1c < 7,0 %",
    short_description: "Andel pasienter med HbA1c < 7,0 %",
    rname: "barnediabetes",
    level_green: 0.4,
    level_yellow: 0.2,
    level_direction: 1,
  });
  const { container } = render(
    <QueryParamProvider adapter={NextAdapter}>
      <table>
        <tbody>
          <IndicatorRow
            context={{ context: "caregiver", type: "ind" }}
            description={descr}
            indicatorData={[data]}
            treatmantYear={2017}
          />
        </tbody>
      </table>
    </QueryParamProvider>
  );
  expect(container).toMatchSnapshot();
});

it("renders with low N", () => {
  const data = buildIndicator({
    unit_name: "UNN",
    denominator: 6,
  });
  const descr = buildDescription({
    title: "qwerty",
    short_description: "Andel pasienter med qwerty",
    min_denominator: 10,
  });
  const { container } = render(
    <QueryParamProvider adapter={NextAdapter}>
      <table>
        <tbody>
          <IndicatorRow
            context={{ context: "caregiver", type: "ind" }}
            description={descr}
            indicatorData={[data]}
            treatmantYear={2017}
            unitNames={["UNN"]}
          />
        </tbody>
      </table>
    </QueryParamProvider>
  );
  expect(container).toMatchSnapshot();
});

it("renders with low dg", () => {
  const data = buildIndicator({
    unit_name: "UNN",
    dg: 0.5,
  });
  const descr = buildDescription({
    title: "qwerty",
    short_description: "Andel pasienter med qwerty",
  });
  const { container } = render(
    <QueryParamProvider adapter={NextAdapter}>
      <table>
        <tbody>
          <IndicatorRow
            context={{ context: "caregiver", type: "ind" }}
            description={descr}
            indicatorData={[data]}
            treatmantYear={2017}
            unitNames={["UNN"]}
          />
        </tbody>
      </table>
    </QueryParamProvider>
  );
  expect(container).toMatchSnapshot();
});

it("renders with missing data", () => {
  const data = buildIndicator({
    unit_name: "qwerty",
  });
  const descr = buildDescription({
    title: "qwerty",
    short_description: "Andel pasienter med qwerty",
  });
  const { container } = render(
    <QueryParamProvider adapter={NextAdapter}>
      <table>
        <tbody>
          <IndicatorRow
            context={{ context: "caregiver", type: "ind" }}
            description={descr}
            indicatorData={[data]}
            treatmantYear={2017}
            unitNames={["UNN"]}
          />
        </tbody>
      </table>
    </QueryParamProvider>
  );
  expect(container).toMatchSnapshot();
});
