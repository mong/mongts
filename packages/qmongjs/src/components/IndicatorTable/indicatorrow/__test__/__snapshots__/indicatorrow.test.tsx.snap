// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with low N 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="testdata  null  indicator"
        data-testid="indicatorrow_testdata"
        id="testdata  null"
      >
        <td
          class="quality_indicator"
        >
          <div
            class="quality_indicator_name"
          >
            <h1>
              qwerty
            </h1>
          </div>
          <div
            class="qi_long_description"
          >
            <p>
              Andel pasienter med qwerty
            </p>
          </div>
        </td>
        <td
          aria-label="Gitt indikator for valgt behandler vises ikke fordi antall observasjoner er lavere enn 10."
          class="nationally"
          data-mui-internal-clone-element="true"
        >
          <div
            class="maskedIndicator"
          >
            Lav N
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders with low achievement level and not complete year 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="barnediabetes_hba1c_ge_9  barnediabetes  indicator"
        data-testid="indicatorrow_barnediabetes_hba1c_ge_9"
        id="barnediabetes_hba1c_ge_9  barnediabetes"
      >
        <td
          class="quality_indicator"
        >
          <div
            class="quality_indicator_name"
          >
            <h1>
              HbA1c ≥ 9,0 %
            </h1>
          </div>
          <div
            class="qi_long_description"
          >
            <p>
              Andel pasienter med HbA1c &gt; 9,0
            </p>
          </div>
          <div
            class="not_complete_year"
          >
            <h4>
              Data for valgt år er ikke komplett. Siste komplette år er
               
              2018
            </h4>
          </div>
        </td>
        <td
          class="nationally "
        >
          <div
            aria-label="Achieved level L"
            class="level"
          >
            <h4>
              11 %
              <i
                style="padding-left: 0.2em;"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(227, 7, 19); font-size: 1.2rem;"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                  />
                </svg>
              </i>
            </h4>
          </div>
          <div
            class="summary"
          >
            300 av 2775
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders with low dg 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="testdata  null  indicator"
        data-testid="indicatorrow_testdata"
        id="testdata  null"
      >
        <td
          class="quality_indicator"
        >
          <div
            class="quality_indicator_name"
          >
            <h1>
              qwerty
            </h1>
          </div>
          <div
            class="qi_long_description"
          >
            <p>
              Andel pasienter med qwerty
            </p>
          </div>
        </td>
        <td
          aria-label="Resultater fra enhet med dekningsgrad lavere enn 60 % vises ikke.       Dekningsgrad angir andel pasienter eller hendelser som registreres, i forhold til antall som skal registreres for gitt indikator.  Om en indikatoren har lav dekningsgrad, er det fare for at utvalget  er skjevt slik at resultatene kan være misvisende og medføre  feilaktige konklusjoner."
          class="nationally"
          data-mui-internal-clone-element="true"
        >
          <div
            class="maskedIndicator"
          >
            Lav dg
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders with missing data 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="testdata  null  indicator"
        data-testid="indicatorrow_testdata"
        id="testdata  null"
      >
        <td
          class="quality_indicator"
        >
          <div
            class="quality_indicator_name"
          >
            <h1>
              qwerty
            </h1>
          </div>
          <div
            class="qi_long_description"
          >
            <p>
              Andel pasienter med qwerty
            </p>
          </div>
        </td>
        <td
          aria-label="Gitt indikator har ingen data for valgt behandler. Dette er enten fordi behandler ikke har rapportert data, eller fordi antall observasjoner er lavere enn 5."
          class="selected_unit"
          data-mui-internal-clone-element="true"
        >
          <div
            class="maskedIndicator"
          >
            Ingen Data
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders with no achievement level on indicator 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="barnediabetes_hba1c_lt_7  barnediabetes  indicator"
        data-testid="indicatorrow_barnediabetes_hba1c_lt_7"
        id="barnediabetes_hba1c_lt_7  barnediabetes"
      >
        <td
          class="quality_indicator"
        >
          <div
            class="quality_indicator_name"
          >
            <h1>
              HbA1c &lt; 7,0 %
            </h1>
          </div>
          <div
            class="qi_long_description"
          >
            <p>
              Andel pasienter med HbA1c &lt; 7,0 %
            </p>
          </div>
          <div
            class="desired_target_level"
          >
            <h4>
              ØNSKET MÅLNIVÅ: 
              ≥
               
              40 %
            </h4>
          </div>
        </td>
        <td
          class="nationally "
        >
          <div
            aria-label="Achieved level undefined"
            class="level"
          >
            <h4>
              20 %
              <i
                style="padding-left: 0.2em;"
              />
            </h4>
          </div>
          <div
            class="summary"
          >
            532 av 2676
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
